### PR TITLE
Fix nullpointer when using RealmTransformer

### LINKED ISSUE #6082: Gradle build error:  org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':sdk:transformClassesWithRealmTransformerForDebug'.
#### Goal

On realm-gradle-plugin:5.3.0, the gradle seems to work fine but as I upgrade to realm-gradle-plugin:5.4.0, the build creation is failing.
#### Expected Results

> ?

#### Actual Results

>  org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':sdk:transformClassesWithRealmTransformerForDebug'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:100)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:70)
        at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:60)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:97)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:87)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:248)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:241)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:230)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:123)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:79)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:104)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:98)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:626)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:581)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:98)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: java.lang.NullPointerException
        at io.realm.transformer.Utils.getTargetSdk(Utils.java:67)
        at io.realm.transformer.RealmTransformer.sendAnalytics(RealmTransformer.kt:154)
        at io.realm.transformer.RealmTransformer.exitTransform(RealmTransformer.kt:116)
        at io.realm.transformer.RealmTransformer.transform(RealmTransformer.kt:111)
        at com.android.build.api.transform.Transform.transform(Transform.java:288)
        at com.android.build.gradle.internal.pipeline.TransformTask$2.call(TransformTask.java:221)
        at com.android.build.gradle.internal.pipeline.TransformTask$2.call(TransformTask.java:217)
        at com.android.builder.profile.ThreadRecorder.record(ThreadRecorder.java:102)
        at com.android.build.gradle.internal.pipeline.TransformTask.transform(TransformTask.java:212)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        at org.gradle.api.internal.project.taskfactory.IncrementalTaskAction.doExecute(IncrementalTaskAction.java:46)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:121)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:110)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:92)
        ... 29 more
#### Steps & Code to Reproduce

> Describe your current debugging efforts.

#### Code Sample

```java

> Your code here. Bigger samples should ideally be as separate Android Studio project, 
> in gists/repositories or privately at help@realm.io)

```

#### Version of Realm and tooling
Realm version(s): ?

Realm sync feature enabled: yes/no

Android Studio version: 3.1.3

Which Android version and device:

### PR DESCRIPTION
**Pull request by cmelchior**
Closes #6082 

Converts the util functions to Kotlin and added extra crash safety when sending analytics (basically, we should never crash no matter what goes wrong). Now all exceptions are just silently logged to the debug log.

### CODE REVIEW

File: realm-transformer/src/main/kotlin/io/realm/transformer/RealmTransformer.kt [hunk 1]
```diff
@@ -123,41 +125,46 @@ class RealmTransformer(val project: Project) : Transform() {
      * @param inputModelClasses a list of ctClasses describing the Realm models
      */
     private fun sendAnalytics(inputs: Collection<TransformInput>, outputModelClasses: Set<CtClass>) {
-        val disableAnalytics: Boolean = "true".equals(System.getenv()["REALM_DISABLE_ANALYTICS"])
-        if (inputs.isEmpty() || disableAnalytics) {
-            // Don't send analytics for incremental builds or if they have ben explicitly disabled.
-            return
-        }
+        try {
+            val disableAnalytics: Boolean = "true".equals(System.getenv()["REALM_DISABLE_ANALYTICS"])
```
Comment (nhachicha) [inline realm-transformer/src/main/kotlin/io/realm/transformer/RealmTransformer.kt]: maybe use `equalsIgnorecase` for extra safety

File: realm-transformer/src/main/kotlin/io/realm/transformer/RealmTransformer.kt [hunk 2]
```diff
@@ -123,41 +125,46 @@ class RealmTransformer(val project: Project) : Transform() {
...
+            if (inputs.isEmpty() || disableAnalytics) {
+                // Don't send analytics for incremental builds or if they have ben explicitly disabled.
```
Comment (nhachicha) [inline realm-transformer/src/main/kotlin/io/realm/transformer/RealmTransformer.kt]: been

File: realm-transformer/src/main/kotlin/io/realm/transformer/ext/ProjectExt.kt [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+/**
+ * Returns the `bootClasspath` for this project
+ */
+fun Project.getBootClasspath(): List<File> {
+    return getAndroidExtension(this).bootClasspath ?: listOf()
+}
+
+private fun getAndroidExtension(project: Project): BaseExtension {
```
Comment (nhachicha) [inline realm-transformer/src/main/kotlin/io/realm/transformer/ext/ProjectExt.kt]: is it worth inlining this since it's called often above?

Comment (cmelchior) [reply realm-transformer/src/main/kotlin/io/realm/transformer/ext/ProjectExt.kt]: Doubtful. The above functions are only called once or twice when the transformer is running.

**Pull request merged**